samples/templates/default-azuredeploy.json

```diff
@@ -216,25 +259,28 @@
                 }
             },
             "dependsOn": [
-                "[if(empty(parameters('appServicePlanResourceGroup')), variables('appServicePlanResourceId'), resourceId('Microsoft.Sql/servers', variables('serviceName')))]"
+                "[resourceId('Microsoft.Web/serverfarms/', variables('appServicePlanName'))]"
             ],
             "resources": [
                 {
                     "apiVersion": "2015-08-01",
-                    "name": "appsettings",
                     "type": "config",
+                    "name": "appsettings",
                     "dependsOn": [
                         "[variables('appServiceResourceId')]",
-                        "[if(variables('deployAppInsights'),concat('Microsoft.Insights/components/', variables('appInsightsName')),resourceId('Microsoft.KeyVault/vaults', variables('serviceName')))]",
+                        "[variables('sqlServerConnectionStringResourceId')]",
+                        "[variables('azureStorageConnectionStringResourceId')]",
+                        "[variables('appInsightsInstrumentationKeyResourceId')]",
+                        "[variables('functionAccessKeyResourceId')]",
                         "[concat(variables('appServiceResourceId'), '/siteextensions/AspNetCoreRuntime.5.0.x64')]",
                         "[concat(variables('appServiceResourceId'), '/extensions/MSDeploy')]"
                     ],
-                    "properties": "[if(variables('deployAppInsights'), union(variables('combinedDicomServerConfigProperties'), json(concat('{\"ApplicationInsights:InstrumentationKey\": \"', reference(concat('Microsoft.Insights/components/', variables('appInsightsName'))).InstrumentationKey, '\"}'))), variables('combinedDicomServerConfigProperties'))]"
+                    "properties": "[union(if(variables('deployAppInsights'), json(concat('{\"ApplicationInsights__InstrumentationKey\": \"', concat('@Microsoft.KeyVault(SecretUri=', reference(variables('appInsightsInstrumentationKeyResourceId')).secretUriWithVersion, ')'), '\"}')), json('{}')), json(concat('{\"BlobStore__ConnectionString\": \"', concat('@Microsoft.KeyVault(SecretUri=', reference(variables('azureStorageConnectionStringResourceId')).secretUriWithVersion, ')'), '\"}')), json(concat('{\"DicomFunctions__FunctionAccessKey\": \"', concat('@Microsoft.KeyVault(SecretUri=', reference(variables('functionAccessKeyResourceId')).secretUriWithVersion, ')'), '\"}')), json(concat('{\"SqlServer__ConnectionString\": \"', concat('@Microsoft.KeyVault(SecretUri=', reference(variables('sqlServerConnectionStringResourceId')).secretUriWithVersion, ')'), '\"}')), variables('combinedDicomServerConfigProperties'))]"
```

Comment:
Why did this change? Previously it would stuff these variables into keyvault and on startup it would use the keyvault setting provider to read them in.

Reply:
I need the Azure Storage Connection string as an environment variable for the Azure Function Runtime, as that's where it stores metadata about the functions and their execution. And because we already store it in keyvault, I thought I'd reference it using a "key vault reference" within the app settings. Once I did that for the Azure Function App, I thought I'd normalize the behavior across our two applications.

Reply:
Yeah, but there is a level of indirection because I am using "Key Vault References." To the web app, the values will be present inside of environment variables, but users won't be able to see the values of these variables in the Azure Portal unless they access them through Key Vault.